bgpd/bgp_mac.c

```diff
@@ -284,11 +284,13 @@ static void bgp_mac_remove_ifp_internal(struct bgp_self_mac *bsm, char *ifname,
 	}
 
 	if (bsm->ifp_list->count == 0) {
+		struct ethaddr mac = *macaddr;
+
 		hash_release(bm->self_mac_hash, bsm);
 		list_delete(&bsm->ifp_list);
 		XFREE(MTYPE_BSM, bsm);
 
-		bgp_mac_rescan_all_evpn_tables(macaddr);
+		bgp_mac_rescan_all_evpn_tables(&mac);
```

Comment:
what if we move bgp_mac_rescan_all_evpn_tables() call to before XFREE and/or list_delete instead of making a copy and use it after free ?

Reply:
I cannot say exactly what bgp_mac_rescan_all_evpn_tables does.  I would prefer to not have to think that through.  not sure we are saving much here though by making that change.

Reply:
I can check that for you. Not about saving anything, trying to avoid the usage after it is freed.

Reply:
@donaldsharp I checked the code flow for function bgp_mac_rescan_all_evpn_tables and usage of passed macaddr. 

bgp_mac_rescan_all_evpn_tables  ==> bgp_mac_rescan_evpn_table  ==> bgp_process_mac_rescan_table == > mac address is used to compare and check if any MAC/IP route is ever advertised and if so marked the rn for further processing. So I don't see any thing wrong to move this call to before XFREE, instead of making a copy of this.

Reply:
I think it makes sense what @srimohans says (just looked over the code what's going on how bgp_mac_remove_ifp_internal() is called from other places).